test/test_common/network_utility.cc

```diff
@@ -116,10 +116,24 @@ Address::InstanceConstSharedPtr getCanonicalLoopbackAddress(Address::IpVersion v
   return Network::Utility::getIpv6LoopbackAddress();
 }
 
-Address::InstanceConstSharedPtr getAnyAddress(const Address::IpVersion version) {
+// There is no portable way to initialize sockaddr_in6 with a static initializer, do it with a
+// helper function instead
+static sockaddr_in6 v6any_() {
+  sockaddr_in6 v6any = {};
+  v6any.sin6_family = AF_INET6;
+  v6any.sin6_addr = in6addr_any;
+
+  return v6any;
+}
+
+Address::InstanceConstSharedPtr getAnyAddress(const Address::IpVersion version, bool v4_compat) {
   if (version == Address::IpVersion::v4) {
     return Network::Utility::getIpv4AnyAddress();
   }
+  if (v4_compat) {
+    static Address::InstanceConstSharedPtr any(new Address::Ipv6Instance(v6any_(), false));
```

Comment:
Can you add some internal comments here explaining how this (maybe using ASCII address notation) differs from `Network::Utility::getIpv6AnyAddress()` below?

Reply:
Added this:
```
    // This will return an IPv6 ANY address ("[::]:0") like the getIpv6AnyAddress() below, but
    // with the internal 'v6only' member set to false. This will allow a socket created from this
    // address to accept IPv4 connections. IPv4 connections received on IPv6 sockets will have
    // Ipv4-mapped IPv6 addresses, which we will then internally interpret as IPv4 addresses so
    // that, for example, access logging will show IPv4 address format for IPv4 connections even
    // if they were received on an IPv6 socket.
```